cmds/core/watchdog/watchdog_linux.go

```diff
@@ -62,71 +62,76 @@ func runCommand() error {
 		}
 	}()
 
-	switch flag.Arg(0) {
+	switch args[0] {
 	case "keepalive":
-		if flag.NArg() != 1 {
-			usage()
+		if len(args) != 1 {
+			return errUsage
 		}
 		if err := wd.KeepAlive(); err != nil {
 			return err
 		}
 	case "settimeout":
-		if flag.NArg() != 2 {
-			usage()
+		if len(args) != 2 {
+			return errUsage
 		}
-		d, err := time.ParseDuration(flag.Arg(1))
+		d, err := time.ParseDuration(args[1])
 		if err != nil {
 			return err
 		}
 		if err := wd.SetTimeout(d); err != nil {
 			return err
 		}
 	case "setpretimeout":
-		if flag.NArg() != 2 {
-			usage()
+		if len(args) != 2 {
+			return errUsage
 		}
-		d, err := time.ParseDuration(flag.Arg(1))
+		d, err := time.ParseDuration(args[1])
 		if err != nil {
 			return err
 		}
 		if err := wd.SetPreTimeout(d); err != nil {
 			return err
 		}
 	case "gettimeout":
-		if flag.NArg() != 1 {
-			usage()
+		if len(args) != 1 {
+			return errUsage
 		}
 		i, err := wd.Timeout()
 		if err != nil {
 			return err
 		}
 		fmt.Println(i)
 	case "getpretimeout":
-		if flag.NArg() != 1 {
-			usage()
+		if len(args) != 1 {
+			return errUsage
 		}
 		i, err := wd.PreTimeout()
 		if err != nil {
 			return err
 		}
 		fmt.Println(i)
 	case "gettimeleft":
-		if flag.NArg() != 1 {
-			usage()
+		if len(args) != 1 {
+			return errUsage
 		}
 		i, err := wd.TimeLeft()
 		if err != nil {
 			return err
 		}
 		fmt.Println(i)
 	default:
-		return fmt.Errorf("unrecognized command: %q", flag.Arg(0))
+		return fmt.Errorf("unrecognized command: %q", args[0])
```

Comment:
There is so much duplicated code here I keep thinking this could be done with a table. Maybe later.

Reply:
yeah, but let's add some tests firsts :)